IM_calculation/IM/structural_response.py

```diff
@@ -0,0 +1,436 @@
+
+    total_accel = ground_accel + rel_accel
+
+    return (
+        displacement,
+        slope,
+        moment,
+        storey_moment,
+        shear,
+        storey_shear,
+        load,
+        ground_accel,
+        rel_accel,
+        total_accel,
+    )
+
+
+def extract_peak_structural_response(
+    displacement,
+    slope,
+    moment,
+    storey_moment,
+    shear,
+    storey_shear,
+    load,
+    ground_accel,
+    rel_accel,
+    total_accel,
+):
+    # extracts the peak response structural response from the time history response matrices
+
+    displacement_peak = get_peak_values(displacement)
+    slope_peak = get_peak_values(slope)
+    moment_peak = get_peak_values(moment)
+    storey_moment_peak = get_peak_values(storey_moment)
+    shear_peak = get_peak_values(shear)
+    storey_shear_peak = get_peak_values(storey_shear)
+    load_peak = get_peak_values(load)
+    ground_accel_peak = get_peak_values(ground_accel)
+    rel_accel_peak = get_peak_values(rel_accel)
+    total_accel_peak = get_peak_values(total_accel)
+
+    # shear is effectively zero at top, set it to zero
+    shear_peak[-1] = 0.0
+    storey_shear_peak[-1] = 0.0
+    # moment is effectively zero at top, set it to zero
+    moment_peak[-1] = 0.0
+    storey_moment_peak[-1] = 0.0
+
+    return (
+        displacement_peak,
+        slope_peak,
+        moment_peak,
+        storey_moment_peak,
+        shear_peak,
+        storey_shear_peak,
+        load_peak,
+        ground_accel_peak,
+        rel_accel_peak,
+        total_accel_peak,
+    )
+
+
+def get_peak_values(all_values):
+    return np.max(np.abs(all_values), axis=1)
+
+
+def get_structural_response(waveform, dt, storey, t1, alpha, c):
+    # 3a calculate the vibration periods of the structure
+    gamma, vibration_period = calculate_vibration_periods(alpha, t1)
+    # 3b calculate the mode shapes of the structure
+    phi, phi_1, phi_2, phi_3, phi_4, participation_factor = calculate_mode_shapes(
+        alpha, gamma, storey
+    )
+    # 3c calculate structural response
+    disp, slope, moment, storey_moment, shear, storey_shear, load, ground_accel, rel_accel, total_accel = calculate_structural_response(
+        vibration_period,
+        waveform,
+        dt,
+        storey,
+        participation_factor,
+        phi,
+        phi_1,
+        phi_2,
+        phi_3,
+        phi_4,
+        alpha,
+        c=c,
+    )
+    # 3d extract peak structural response values from each earthquake ground motion record
+    disp_peak, slope_peak, moment_peak, storey_moment_peak, shear_peak, storey_shear_peak, load_peak, ground_accel_peak, rel_accel_peak, total_accel_peak = extract_peak_structural_response(
+        disp,
+        slope,
+        moment,
+        storey_moment,
+        shear,
+        storey_shear,
+        load,
+        ground_accel,
+        rel_accel,
+        total_accel,
+    )
+    return disp_peak, moment_peak
```

Comment:
Need to find out which measures we actually want to store, From memory it was disp and slope, but need to double check.

Reply:
Return disp_peak, slope_peak, storey_shear_peak, and total_accel_peak.